pkgs/swiftgen/swift2objc/lib/src/transformer/transformers/transform_method.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:swift2objc/src/ast/_core/shared/parameter.dart';
+import 'package:swift2objc/src/ast/_core/shared/referred_type.dart';
+import 'package:swift2objc/src/ast/declarations/compounds/class_declaration.dart';
+import 'package:swift2objc/src/transformer/transform.dart';
+import 'package:swift2objc/src/transformer/transformers/transform_referred_type.dart';
+
+ClassMethodDeclaration tranformMethod(
+  ClassMethodDeclaration originalMethod,
+  TransformationMap transformationMap,
+) {
+  final transformedParams = originalMethod.params
+      .map((param) => _tranformParamter(param, transformationMap))
+      .toList();
+
+  final transformedMethod = ClassMethodDeclaration(
+    id: originalMethod.id,
+    name: originalMethod.name,
+    returnType: transformReferredType(
+      originalMethod.returnType,
+      transformationMap,
+    ),
+    params: transformedParams,
+    hasObjCAnnotation: true,
+  );
+
+  transformedMethod.statements = <String>[
+    _generateMethodCall(originalMethod, transformedMethod)
+  ];
+
+  transformationMap[originalMethod] = transformedMethod;
+
+  return transformedMethod;
+}
+
+String _generateMethodCall(
+  ClassMethodDeclaration originalMethod,
+  ClassMethodDeclaration transformedMethod,
+) {
+  final arguments = <String>[];
+
+  for (var i = 0; i < originalMethod.params.length; i++) {
+    final originalParam = originalMethod.params[i];
+    final transformedParam = transformedMethod.params[i];
+
+    String methodCallArg =
+        "${originalParam.name}: ${transformedParam.internalName ?? transformedParam.name}";
+
+    final transformedType = transformedParam.type;
+    if (transformedType is DeclaredType) {
+      final typeDeclaration = transformedType.declaration;
+      if (typeDeclaration is ClassDeclaration && typeDeclaration.isWrapper) {
+        methodCallArg += ".${typeDeclaration.originalInstance!.name}";
+      }
+    }
+
+    arguments.add(methodCallArg);
+  }
+
+  return "originalInstance.${originalMethod.name}(${arguments.join(", ")})";
```

Comment:
The result may need to be wrapped, in the same way that the args are unwrapped.

Reply:
hmmm I think this will need us to work on class initializers too for this to make sense. Wdyt?

Reply:
I don't think so? You've got class `Foo` and you're wrapping it in `FooWrapper`. When a wrapped method returns a `Foo`, you should wrap it in a `FooWrapper`. So you need the `FooWrapper`'s constructor, but you don't need to understand the `Foo`'s constructors.

If I'm missing something, and you need to implement class initializers to make this work, then don't worry about it for now. Finish codegen first so you get the pipeline working end-to end. Then when you start adding support for more features, class initializers should be at the top of that list, and you can fix this problem then.

Reply:
Yup ur right. I still need to make a constructor with the wrapped instance as a parameter, but I think it's easy enough to do so I'll include it in this PR.

Reply:
Done. Have a look.